src/frameworks/compose/discover/index.ts

```diff
@@ -1,26 +1,52 @@
-import { AppSpec } from "../interfaces";
+import { Runtime, FileSystem, FrameworkSpec, RuntimeSpec, FrameworkHooks } from "./types";
+import { NodejsRuntime } from "./runtime/node";
+import { FirebaseError } from "../../../error";
+import { AppBundle } from "../interfaces";
 
 /**
  * Discover framework in the given project directory
  */
-export function discover(): AppSpec {
-  return {
-    baseImage: "us-docker.pkg.dev/firestack-build/test/run:latest",
-    environmentVariables: {
-      NODE_ENV: "PRODUCTION",
-    },
-    installCommand: "npm install",
-    buildCommand: "npm run build",
-    startCommand: "npm run start",
+const availableRuntimes: Runtime[] = [new NodejsRuntime()];
+
+/**
+ *
+ */
+export async function discover(
+  fs: FileSystem,
+  allFrameworkSpecs: FrameworkSpec[]
+): Promise<RuntimeSpec> {
+  try {
+    let discoveredRuntime = undefined;
+    for (const runtime of availableRuntimes) {
+      if (await runtime.match(fs)) {
+        if (!discoveredRuntime) {
+          discoveredRuntime = runtime;
+        } else {
+          throw new FirebaseError("Multiple runtimes discovered for the codebase");
```

Comment:
didn't we try to update the error message to be more helpful 🤔  I forgot if we ever landed on what this should say instead or we decided to punt on it.

Reply:
Adding what runtimes are found may help user? WDYT